sdlib/d/gc/tcache.d

```diff
@@ -54,6 +64,39 @@ public:
 		pd.arena.free(emap, pd, ptr);
 	}
 
+	// Determine whether given alloc is appendable
+	bool is_appendable(void* ptr) {
+		return maybeGetAppendableExtent(ptr) !is null;
+	}
+
+	// Get the current fill of an appendable alloc.
+	// If the alloc is not appendable, returns 0.
+	size_t get_appendable_fill(void* ptr) {
```

Comment:
This won't work with appendable pointer. This API is still a more or less random set of features that do not seems to be connected with the actual use case.

The actual use case is that D has slices, and slices can be appended to, and reallocations needs to be avoided when unnecessary.

I don't know how to do any of this with the current API, considering it doesn't work with interior pointers.

So, what happens when I allocate a slice? What happens when I want to append to a slice? What happens when the slice is not on the GC heap? This is what matters here. I have no API do this.